components/ILIAS/Mail/classes/Folder/MailFilterUI.php

```diff
@@ -0,0 +1,149 @@
+        private readonly bool $lucene_enabled,
+        private readonly MailFolderData $folder,
+        private readonly Factory $ui_factory,
+        private readonly ilUIFilterService $filter_service,
+        private readonly ilLanguage $lng,
+        private readonly DateTimeZone $user_time_zone
+    ) {
+        $inputs = [];
+        if ($this->folder->hasIncomingMails()) {
+            $inputs['sender'] = $this->ui_factory->input()->field()->text($this->lng->txt('mail_filter_sender'));
+        } else {
+            $inputs['recipients'] = $this->ui_factory->input()->field()->text($this->lng->txt('mail_filter_recipients'));
+        }
+
+        $inputs['subject'] = $this->ui_factory->input()->field()->text($this->lng->txt('mail_filter_subject'));
+        $inputs['body'] = $this->ui_factory->input()->field()->text($this->lng->txt('mail_filter_body'));
+
+        if ($this->lucene_enabled) {
+            $inputs['attachment'] = $this->ui_factory->input()->field()->text($this->lng->txt('mail_filter_attach'));
+        }
+
+        $inputs['display'] = $this->ui_factory->input()->field()->multiSelect($this->lng->txt('mail_filter_display'), [
+            'read' => $this->lng->txt('mail_filter_show_read'),
+            'unread' => $this->lng->txt('mail_filter_show_unread'),
+            'user' => $this->lng->txt('mail_filter_show_user_mails'),
+            'system' => $this->lng->txt('mail_filter_show_system_mails'),
+            'with_attachment' => $this->lng->txt('mail_filter_show_with_attachments'),
+            'without_attachment' => $this->lng->txt('mail_filter_show_without_attachment')
+        ]);
+        $inputs['period'] = $this->ui_factory->input()->field()->duration($this->lng->txt('mail_filter_period'))
+                                                               ->withTimezone($this->user_time_zone->getName());
+
+        $this->filter = $this->filter_service->standard(
+            self::class,
+            $this->target_url,
+            //elements
+            $inputs,
+            // initially rendered
+            array_map(fn($value) => true, $inputs),
+            false,
+            false
+        );
+    }
+
+    /**
+     * Get the filter UI component
+     */
+    public function getComponent(): FilterComponent
+    {
+        return $this->filter;
+    }
+
+    /**
+     * Get the user entered filter data
+     */
+    public function getData(): MailFilterData
+    {
+        $data = $this->filter_service->getData($this->filter);
+
+        $is_unread = null;
+        $is_system = null;
+        $has_attachment = null;
+
+        if (is_array($display = $data['display'] ?? null)) {
+            // filter is applied only if one of read/unread is set
+            if (in_array('unread', $display) && !in_array('read', $display)) {
+                $is_unread = true;
+            }
+            if (in_array('read', $display) && !in_array('unread', $display)) {
+                $is_unread = false;
+            }
```

Comment:
This can be simplified to
```suggestion
            if (in_array('read', $display) xor in_array('unread', $display)) {
                $is_unread = in_array('unread', $display);
            }
```

same for the 2 following assumptions